src/main/kotlin/no/nav/familie/ba/sak/kjerne/vilkårsvurdering/preutfylling/PreutfyllBosattIRiketService.kt

```diff
@@ -170,3 +201,19 @@ class PreutfyllBosattIRiketService(
             }.tilTidslinje()
     }
 }
+
+private sealed class Delvilkår {
+    abstract val erOppfylt: Boolean
+    abstract val begrunnelse: String
+
+    data class OppfyltDelvilkår(
+        override val begrunnelse: String,
+    ) : Delvilkår() {
+        override val erOppfylt: Boolean = true
+    }
+
+    data object IkkeOppfyltDelvilkår : Delvilkår() {
+        override val erOppfylt: Boolean = false
+        override val begrunnelse: String = ""
+    }
+}
```

Comment:
Hvis `OppfyltDelvilkår` og `IkkeOppfyltDelvilkår` blir flyttet ut av `Delvilkår` slipper man å skrive `Delvilkår.OppfyltDelvilkår` overalt. Og så trenger egentlig ikke `Delvilkår` være en `sealed class` når dere ikke gjør noe pattern matching, den kan være et `interface`.

```suggestion
private interface Delvilkår {
    val erOppfylt: Boolean
    val begrunnelse: String
}

private data class OppfyltDelvilkår(
    override val begrunnelse: String,
) : Delvilkår {
    override val erOppfylt: Boolean = true
}

private data object IkkeOppfyltDelvilkår : Delvilkår {
    override val erOppfylt: Boolean = false
    override val begrunnelse: String = ""
}
```